From 9ac3812ec97c4b8d5d9ae0d27585a2319337a0f0 Mon Sep 17 00:00:00 2001
From: Bui Quang Minh <minhquangbui99@gmail.com>
Date: Sat, 12 Feb 2022 16:00:39 +0700
Subject: [PATCH 133/249] ci: skip MAP_HUGETLB tests in stream test

Currently, hugetlb mappings is not premapped so in the restore content phase, we
skip page read these pages, enqueue the iovec for later reading in restorer and
eventually close the page read. However, image-streamer expects the whole image
to be read and the image is not re-opened, sent twice. These MAP_HUGETLB test
cases will result in EPIPE error. Temporarily disable these test cases for now.

Signed-off-by: Bui Quang Minh <minhquangbui99@gmail.com>
---
 scripts/ci/run-ci-tests.sh | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/scripts/ci/run-ci-tests.sh b/scripts/ci/run-ci-tests.sh
index d0cd55f7c..50216634e 100755
--- a/scripts/ci/run-ci-tests.sh
+++ b/scripts/ci/run-ci-tests.sh
@@ -63,8 +63,15 @@ ci_prep () {
 
 test_stream() {
 	# Testing CRIU streaming to criu-image-streamer
+
+	# FIXME: Currently, hugetlb mappings is not premapped, so in the restore content
+	# phase, we skip page read these pages, enqueue the iovec for later reading in
+	# restorer and eventually close the page read. However, image-streamer expects the
+	# whole image to be read and the image is not reopened, sent twice. These MAP_HUGETLB
+	# test cases will result in EPIPE error at the moment.
+	STREAM_TEST_EXCLUDE="-x maps09 -x maps10"
 	# shellcheck disable=SC2086
-	./test/zdtm.py run --stream -p 2 --keep-going -a $ZDTM_OPTS
+	./test/zdtm.py run --stream -p 2 --keep-going -a $STREAM_TEST_EXCLUDE $ZDTM_OPTS
 }
 
 print_header() {
-- 
2.35.1

